### ISSUE TITLE
Can only Toggle no_straight_on Turn Restriction Once

### ISSUE THREAD

**Issue by zlavergne** [openstreetmap/iD#4928]
I ran into an issue where I can only toggle a turn restriction once in the new Restriction editor. I discovered the bug because I accidentally hit the `no_straight` but I couldn't click through to reset it. 

![turn_restriction](https://user-images.githubusercontent.com/9849118/37787983-e5196082-2dc5-11e8-96d3-4a6a6b54e686.gif)

I cannot undo the `no_straight` on any of the ways I could add it to. 

[Here's the way we're looking at.](https://www.openstreetmap.org/way/28598125#map=19/-34.34638/-70.84229)

**Comment by bhousel** [comment]
Hey @zlavergne which Browser/OS did you see this on?
I can't seem to reproduce on MacOS with Chrome 65.0.3325.162

**Comment by zlavergne** [comment]
I'm on MacOS and Chrome 65.0.3325.181 
I'm getting this error from the console: 
```
id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:119 Uncaught Error: entity r-1 not found
    at coreGraph$$1.entity (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:119)
    at id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:21
    at Array.<anonymous> (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:22)
    at n (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:20)
    at t (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:20)
    at Object.perform (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:20)
    at Object.perform (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:21)
    at SVGSVGElement.a (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:18)
    at SVGSVGElement.<anonymous> (id-18ccfbbf4674e7fcefc636d54df5ee8f3363dadd80070ea544340f99000d0bb7.js:1)
```

**Comment by zlavergne** [comment]
Update: It seems to be an issue involving the zoom level. The toggle works fine when on the same way when the window is zoomed in. In the case of the way I sent above, the problem happens at 145ft or higher. 
For that intersection, when you get to 115ft, you are viewing the most you can view and when you change the zoom level, I begin to get the error for select ways.

**Comment by bhousel** [comment]
Thanks @zlavergne I was able to reproduce and debug the issue once I increased the max distance to 145ft.

Here's what's going on (it's weird)..

You're trying to cycle from `no_straight_on` to `only_straight_on`.  iD knows that when you switch to an `only_`, it should remove ALL the `no_` restrictions from the FROM.  So it looks any `no_` restrictions that it can find, searching up to 2 via ways (regardless of how you set the slider, they all need to go away).

The most obvious one it finds (the one you clicked on and that you are removing) is :
`FROM w-2 VIA node TO w-1`, which it identifies as `r-1`
But it found another one too:
`FROM w-2 VIA way VIA way TO w-1`, which is also identifies as... `r-1`  (that's the bug).

<img width="342" alt="screenshot 2018-03-28 21 09 22" src="https://user-images.githubusercontent.com/38784/38065046-30f729f4-32cf-11e8-9187-29d01668e9d3.png">

To summarize, 
When cycling from `no_` to `only_`, iD tries to remove the `no_` restrictions.
It finds `r-1` twice (it should only find it once).
And it tries to remove it twice (causing the failure).

**Comment by bhousel** [comment]
This is fixed, I'm going to test a bit more..

**Comment by zlavergne** [comment]
Good find! I had a hunch it had something to do with deleting a relation that was already deleted. It was odd that it happened only above a certain zoom level even though the number of objects didn't change.
